ADCoordinator/Classes/DeallocObserver.swift

```diff
@@ -12,7 +12,7 @@ import Foundation
  */
 final class DeallocObserver {
 
-    private let callback: () -> Void
+    var callback: () -> Void
```

Comment:
I don't really like the public var callback.
maybe:

 private var isInvalidated = false
func invalidate() {
isInvalidated = true
}

and then guard !isInvalidated in the deinit ?